TestApp/Task3_1.cs

```diff
@@ -0,0 +1,346 @@
+            float?[] taskDeleteInFloatArray = [13.17263f, 84.83026f, 30.99372f, 59.6616f, 61.46767f, 45.17126f, 61.95522f, 93.7445f, 46.02798f, 14.73467f];
+            long?[] taskDeleteInLongArray = [317949394, 814426843, 450488525, 261198928, 167526774, 353225171, 752564584, 727744147, 394885515, 707128204];
+            string?[] taskDeleteInStringArray = ["1WStS2R7Y", "2cFmNcdw2", "3wzqzZGEl", "4bXn0bF1w", "5vhNmqgP", "6IAt5vR3m", "7ZtBbqKc2", "8sK2QexWg", "9tlzq0bPy", "10KYsEfyj"];
+            char?[] taskDeleteInCharArray = { '0', '1', '2', '3', '4', null, '6', '7', '8', '9' };
+            bool?[] taskDeleteInBoolArray = { true, false, true, false, null, false, true, false, true, false };
+            MyStruct?[] taskDeleteInStructArray = { new(0), new(1), new(2), new(3), new(4), new(5), new(6), new(7), new(8), new(9) };
+            FoodType?[] taskDeleteInEnumArray = { FoodType.Pashtetik, FoodType.Kormik, FoodType.Smakolyk, FoodType.Pashtetik, FoodType.Kormik, FoodType.Smakolyk, FoodType.Pashtetik, FoodType.Kormik, FoodType.Smakolyk, FoodType.Pashtetik };
+
+            TestArray(taskDeleteInIntArray, DeleteUsingForLoop, InsertUsingForLoop, 444);
+            TestArray(taskDeleteInObjectArray, DeleteUsingForeachLoop, InsertUsingForeachLoop, 444);
+            TestArray(taskDeleteInFloatArray, DeleteUsingWhileLoop, InsertUsingWhileLoop, 444.4444f);
+            TestArray(taskDeleteInLongArray, DeleteUsingForLoop, InsertUsingForLoop, 444444444);
+            TestArray(taskDeleteInStringArray, DeleteUsingForeachLoop, InsertUsingForeachLoop, "4444444");
+            TestArray(taskDeleteInCharArray, DeleteUsingForLoop, InsertUsingWhileLoop, 'X');
+            TestArray(taskDeleteInBoolArray, DeleteUsingWhileLoop, InsertUsingForLoop, false);
+            TestArray(taskDeleteInStructArray, DeleteUsingForeachLoop, InsertUsingForeachLoop, new(444));
+            TestArray(taskDeleteInEnumArray, DeleteUsingForLoop, InsertUsingWhileLoop, FoodType.Kormik);
+        }
+
+        private static void TestArray<T>(T[] array, Func<T[], int[], T[]> delete, Func<T[], T, int, T[]> insert, T elementToInsert)
+        {
+            // Permanently delete elements [5],[6],[7]
+            int[] indicesToDelete = [5, 6, 7];
+            Console.WriteLine($"Permanently deleting elements at {indicesToDelete} positions: ");
+            
+            var arrayAfterDeletion = delete(array, indicesToDelete);
+            PrintArray(array, arrayAfterDeletion);
+
+            int indexToInsert = 4;
+            Console.WriteLine($"Inserting new element to {indexToInsert} position:");
+            // Insert to [4] new element
+            var arrayAfterInsertion = insert(arrayAfterDeletion, elementToInsert, indexToInsert);
+            PrintArray(arrayAfterDeletion, arrayAfterInsertion);
+        }
+
+
+        private static T[] DeleteUsingLinq<T>(T[] array, int[] at)
```

Comment:
get rid of int[] at

Reply:
animals = [Cat, Dog, Parrot, Rabbit]
del Parrot
det(animals, [1,2,3,4]) - no
only det(animals)
for all cycles